packages/flask/src/UICountryPicker/CountryPicker.tsx

```diff
@@ -0,0 +1,189 @@
+import React from 'react';
+import Fuse from 'fuse.js';
+import { useSafeAreaInsets } from 'react-native-safe-area-context';
+import { View, useWindowDimensions } from 'react-native';
+
+import { UIConstant } from '@tonlabs/uikit.core';
+import { uiLocalized } from '@tonlabs/uikit.localization';
+import { FlatList, UIBottomSheet, UISearchBar } from '@tonlabs/uikit.navigation';
+import {
+    ColorVariants,
+    makeStyles,
+    Theme,
+    TypographyVariants,
+    UILabel,
+    UILinkButton,
+    useTheme,
+} from '@tonlabs/uikit.hydrogen';
+import type { CountriesArray, Country, WrappedCountryPickerProps } from '../types';
+import { CountryPickerRow } from './CountryPickerRow';
+import { ListEmptyComponent } from './ListEmptyComponent';
+
+const COUNTRIES_URL = 'https://uikit.tonlabs.io/countries.json';
+
+const fetchJSON = async () => {
+    const results = await fetch(COUNTRIES_URL);
+    return results.json();
+};
+
+const fuseOptions = {
+    ignoreLocation: true,
+    threshold: 0.4,
+    findAllMatches: true,
+    maxPatternLength: 32,
+    minMatchCharLength: 1,
+    keys: ['name'],
+};
+
+function returnCountryRow({ item }: { item: Country }) {
+    return <CountryPickerRow item={item} />;
+}
+
+type CountryPickerContextType = {
+    loading: boolean;
+    onSelect?(countryCode: string): void;
+};
+
+export const CountryPickerContext = React.createContext<CountryPickerContextType>({
+    loading: true,
+});
+
+export function CountryPicker({
+    onClose,
+    onSelect,
+    visible,
+    banned = [],
+    permitted = [],
+}: WrappedCountryPickerProps) {
+    const height = useWindowDimensions().height;
+
+    const theme = useTheme();
+    const styles = useStyles(theme, height);
+
+    const [loading, setLoading] = React.useState(true);
+
+    const [search, setSearch] = React.useState('');
+    const [countriesList, setCountriesList] = React.useState<CountriesArray>([]);
+    const [filteredList, setFilteredList] = React.useState(countriesList);
+    const fuse = React.useMemo(() => new Fuse(filteredList, fuseOptions), [filteredList]);
+
+    const insets = useSafeAreaInsets();
+    const contentInset = React.useMemo(
+        () => ({
+            top: 0, // without it adds some additional space on iOS
+            bottom: insets.bottom + UIConstant.contentOffset() + UIConstant.buttonHeight(),
+        }),
+        [insets?.bottom],
+    );
+
+    const checkIncludes = React.useCallback((code: string) => {
+        let isPermitted = true;
+        let isBanned = false;
+        if (permitted.length) {
+            isPermitted = permitted.includes(code);
+        }
+        if (banned.length) {
+            isBanned = banned.includes(code);
+        }
+        return isPermitted && !isBanned;
```

Comment:
This seem to be the same in terms of optimization (or even better as it doesn't require two extra .length checks), but looks way shorter and cleaner to me :)
```suggestion
        return permitted.includes(code) && !banned.includes(code);
```